common/remote_object.go

```diff
@@ -78,28 +77,27 @@ func parseRemoteObjectPreview(op *cdpruntime.ObjectPreview) (map[string]any, err
 	return obj, errors.Join(result...)
 }
 
-//nolint:cyclop
 func parseRemoteObjectValue(
-	t cdpruntime.Type, st cdpruntime.Subtype, val string, op *cdpruntime.ObjectPreview,
+	t runtime.Type, st runtime.Subtype, val string, op *runtime.ObjectPreview,
 ) (any, error) {
-	switch t {
-	case cdpruntime.TypeAccessor:
+	switch t { //nolint:exhaustive
+	case runtime.TypeAccessor:
 		return "accessor", nil
-	case cdpruntime.TypeBigint:
-		n, err := strconv.ParseInt(strings.Replace(val, "n", "", -1), 10, 64)
+	case runtime.TypeBigint:
+		n, err := strconv.ParseInt(strings.ReplaceAll(val, "n", ""), 10, 64)
```

Comment:
Not related to the PR, but isn't this an error? I mean that we're using `strconv.ParseInt` for the parsing of `runtime.TypeBigint`? :thinking:

Reply:
This is parsin the BigInt from its string representation, why not directly from the type ... :shrug: 

Arguably one of  the places where I am more questioning why we are doing this at all than anythign else. 

In this case we do first remove `n` which arguably will only be the *last* symbol in the string representation. 

Just changing more than how the code actually does a thing means more chances for error and given the size of the required changes, it already is very likely some corner case isn't handled.

Reply:
That bigInt is coming back from chrome itself in a json payload as a CDP event over the websocket, not from Sobek's runtime, so i think this is a valid way of parsing it from its CDP string representation into one that is needed in k6 browser, and eventually in the test script. The case for this is when we get a piece of data from chrome (e.g. a console message, request, response, etc, containing a JS type of some kind) and the user wants to read that value into the test script.

I may have missed something though?

Reply:
If it comes over CDP as JSON - there is no json representation for bigint in the specification, so I guess this needs this speicifc code. So yeah this is probably is the correct thing to do

Reply:
The confusing part is that how we try to restore the `bigInt`, like instead of `math/bigInt` from GO SDK we treat `bigInt` as an integer.

Consider example:

```golang
package main

import (
	"fmt"
	"math/big"
	"strconv"
	"strings"
)

func main() {
	// A large number string
	bigIntString := "123456789012345678901234567890"

	bigInt := new(big.Int)
	_, success := bigInt.SetString(bigIntString, 10) // Base 10

	if !success {
		fmt.Println("Failed to parse big integer")
		return
	}
	fmt.Println("Parsed big.Int:", bigInt)

	n, err := strconv.ParseInt(strings.ReplaceAll(bigIntString, "n", ""), 10, 64)
	if err != nil {
		fmt.Println("Failed to parse int64 " + err.Error())
		return
	}

	fmt.Println("Parsed N:", n)
}
```

Reply:
I guess you are correct for this, but please open an issue for it as it changes the current behaviour.

Reply:
>I guess you are correct for this, but please open an issue for it as it changes the current behaviour.

What I'm trying to check with @ankur22 (because of lack of my context of browser) if it's the case and we should open the issue

Reply:
Extracted https://github.com/grafana/xk6-browser/issues/1558, if not relevant let's close it

Reply:
Yeah, that's a good point. Worth having the issue open to explore the reason why it's not represented as a go's bigInt.